src/axom/slic/tests/slic_macros.cpp

```diff
@@ -412,6 +413,67 @@ TEST(slic_macros, test_tagged_macros)
   EXPECT_TRUE(slic::internal::is_stream_empty());
 }
 
+//------------------------------------------------------------------------------
+TEST(slic_macros, test_macros_file_output)
+{
+  std::string msgfmt = "<MESSAGE>";
+
+  // GenericOutputStream(std::string stream) and
+  // GenericOutputStream(std::string stream, std::string format) constructors
+  // do not create a a file until macros called, then flushed
+  std::string no_fmt = "file_no_fmt.txt";
+  std::string with_fmt = "file_with_fmt.txt";
+
+  slic::addStreamToAllMsgLevels(new slic::GenericOutputStream(no_fmt));
+
+  slic::addStreamToAllMsgLevels(new slic::GenericOutputStream(with_fmt, msgfmt));
+
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(no_fmt), false);
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(with_fmt), false);
+
+  // streams flushed with no buffered messages, no files created
+  slic::flushStreams();
+
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(no_fmt), false);
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(with_fmt), false);
+
+  // message is buffered but not yet flushed, no files created
+  SLIC_INFO("Test");
+
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(no_fmt), false);
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(with_fmt), false);
+
+  // message has been buffered and now flushed, files are created
+  slic::flushStreams();
+
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(no_fmt), true);
+  EXPECT_EQ(axom::utilities::filesystem::pathExists(with_fmt), true);
+
+  // Verify file contents
+  std::ifstream no_fmt_contents(no_fmt);
+  std::stringstream no_fmt_buffer;
+  no_fmt_buffer << no_fmt_contents.rdbuf();
+
+  std::string no_fmt_expected;
+  no_fmt_expected += "*****\n[INFO]\n\n Test \n\n ";
+  no_fmt_expected += __FILE__;
+  no_fmt_expected += "\n441\n****\n";
+
+  EXPECT_EQ(no_fmt_buffer.str(), no_fmt_expected);
+
+  std::ifstream with_fmt_contents(with_fmt);
+  std::stringstream with_fmt_buffer;
+  with_fmt_buffer << with_fmt_contents.rdbuf();
+
+  EXPECT_EQ(with_fmt_buffer.str(), "Test");
+
+// Cleanup generated files (not working Windows)
```

Comment:
Is it known that this doesn't work on Windows? 
I didn't see an issue for this -- if we don't have one, can you please create one?

Reply:
> I didn't see an issue for this -- if we don't have one, can you please create one?

Opened an issue here: https://github.com/LLNL/axom/issues/1415
I didn't dive too deep into why it wasn't working on Windows for this specific case.
More generally though, there are multiple places where we call `removeFile` in the code but don't check the return value/check if file is really deleted.